pandas/io/formats/format.py

```diff
@@ -1003,6 +1005,121 @@ def to_html(
+
+        .. versionadded:: 1.3.0
+
+        Parameters
+        ----------
+        io : str, path object or file-like object, optional
+            File to write output to. If None, the output is returned as a
+            string.
+        index : bool, optional
+            Whether to include index in XML document.
+        root_name : str, default 'data'
+            The name of root element in XML document.
+        root_name : str, default 'row'
+            The name of row element in XML document.
+        na_rep : str, optional
+            Missing data representation.
+        attr_cols : list-like, optional
+            List of columns to write as attributes in row element.
+            Hierarchical columns will be flattened with underscore
+            delimiting the different levels.
+        elem_cols : list-like, optional
+            List of columns to write as children in row element. By default,
+            all columns output as children of row element. Hierarchical
+            columns will be flattened with underscore delimiting the
+            different levels.
+        namespaces : dict, optional
+            All namespaces to be defined in root element. Keys of dict
+            should be prefix names and values of dict corresponding URIs.
+            Default namespaces should be given empty string key. For
+            example, ::
+
+                namespaces = {'': 'https://example.com'}
+
+        prefix : str, optional
+            Namespace prefix to be used for every element and/or attribute
+            in document. This should be one of the keys in ``namespaces``
+            dict.
+        encoding : str, optional, default 'utf-8'
+            Encoding of the resulting document.
+        xml_declaration : str, optional
+            Whether to include the XML declaration at start of document.
+        pretty_print : bool, optional
+            Whether output should be pretty printed with indentation and
+            line breaks.
+        parser : {'lxml','etree'}, default "lxml"
+            Parser module to use for building of tree. Only 'lxml' and
+            'etree' are supported. With 'lxml', the ability to use XSLT
+            stylesheet is supported. Default parser uses 'lxml'. If
+            module is not installed a warning will raise and process
+            will continue with 'etree'.
+        stylesheet : str, path object or file-like object, optional
+            A URL, file-like object, or a raw string containing an XSLT
+            script used to transform the raw XML output. Script should use
+            layout of elements and attributes from original output. This
+            argument requires ``lxml`` to be installed. Only XSLT 1.0
+            scripts and not later versions is currently supported.
+        """
+
+        from pandas.io.formats.xml import EtreeXMLFormatter, LxmlXMLFormatter
+
+        if parser == "lxml":
+            try:
+                TreeBuilder = LxmlXMLFormatter
```

Comment:
hmm i guess this is ok, but this is pretty much going to warn all the time, is there a reason to have this the default? (lxml)? i would rather simply raise if its not installed and let the user be explict about etree

Reply:
I am re-working this to pass the failed CI tests. Will now use `import_optional_dependency` and raise a warning to user and continue with `etree` as fallback.